Only set up homebrew site functions when `brew` is found

unixorn opened the pull request:
- Make setting and using `HOMEBREW_SITE_FUNCTIONS` only happen when `brew` was found, either in `$PATH` or by searching through `DEFAULT_BREW_PATHS`.

unixorn committed:
Only set up homebrew site functions when `brew` is found

- Make setting and using `HOMEBREW_SITE_FUNCTIONS` only happen when `brew` was found, either in `$PATH` or by searching through `DEFAULT_BREW_PATHS`.
- Unset internal `DEFAULT_BREW_PATHS`, `BREW_PATH` and `HOMEBREW_*` variables after done using them

wintermi commented on zsh-brew.plugin.zsh:
```diff
@@ -15,21 +15,29 @@ if (( ! $+commands[brew] )); then
 
     eval "$("$BREW_PATH" shellenv)"
+    # Tidy up environment
+    unset DEFAULT_BREW_PATHS
```
The unset's are not required as the variables have been set to local, so are scoped to this file only.

unixorn replied:
Thanks for the catch, not sure how I missed that. Removed the `unset` commands

unixorn committed:
Remove `unset`

I missed that the variables are all local, these `unset` commands aren't necessary.

unixorn committed:
Merge branch 'main' into patch-1

wintermi commented:
Thanks @unixorn  😃
